scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1029,12 +1085,70 @@ public void changeJobPriority(JobId jobId, JobPriority priority)
         schedulingService.changeJobPriority(jobId, priority);
     }
 
+    /**
+     * This method blocks until job is actually finished its submission.
+     *
+     * @param jobId
+     */
+    private void blockIfJobIsStillUnderSubmission(JobId jobId) {
+        while (waitForJobsToBeSubmitted.contains(jobId)) {
+            try {
+                waitForJobsToBeSubmitted.wait();
+            } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move this call to "wait()" into a synchronized block to be sure the monitor on "Set" is held. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2273)